### PR TITLE
[python] Allow python sklearn interface's fit() to pass init_model to train()

### PR DESCRIPTION
**Pull request by aaiyer**
Allow python sklearn interface's fit() to pass init_model to train()

### CODE REVIEW

File: python-package/lightgbm/sklearn.py [hunk 1]
```diff
@@ -375,7 +375,8 @@ def fit(self, X, y,
             eval_set=None, eval_names=None, eval_sample_weight=None,
             eval_class_weight=None, eval_init_score=None, eval_group=None,
             eval_metric=None, early_stopping_rounds=None, verbose=True,
-            feature_name='auto', categorical_feature='auto', callbacks=None):
+            feature_name='auto', categorical_feature='auto', 
+            init_model=None, callbacks=None):
```
Comment (StrikerRUS) [inline python-package/lightgbm/sklearn.py]: Please place the new argument either after `init_score=None` as logically correct order, or at the last place (after `callbacks=None`) as for backward compatibility.

Comment (aaiyer) [reply python-package/lightgbm/sklearn.py]: Done, placed it at the end for backward compatibility.

File: python-package/lightgbm/sklearn.py [hunk 2]
```diff
@@ -438,6 +439,8 @@ def fit(self, X, y,
             Large values could be memory consuming. Consider using consecutive integers starting from zero.
             All negative values in categorical features will be treated as missing values.
             The output cannot be monotonically constrained with respect to a categorical feature.
+        init_model : string, Booster or None, optional (default=None)
```
Comment (StrikerRUS) [inline python-package/lightgbm/sklearn.py]: Move according to the new order.

File: python-package/lightgbm/sklearn.py [hunk 3]
```diff
@@ -592,6 +595,7 @@ def _get_meta_data(collection, name, i):
                               evals_result=evals_result, fobj=self._fobj, feval=feval,
                               verbose_eval=verbose, feature_name=feature_name,
                               categorical_feature=categorical_feature,
+                              init_model=init_model,
```
Comment (StrikerRUS) [inline python-package/lightgbm/sklearn.py]: Move according to the new order. Also, please place it at the same line as `callbacks`.

File: python-package/lightgbm/sklearn.py [hunk 4]
```diff
@@ -735,6 +740,7 @@ def fit(self, X, y,
                                        early_stopping_rounds=early_stopping_rounds,
                                        verbose=verbose, feature_name=feature_name,
                                        categorical_feature=categorical_feature,
+                                       init_model=init_model,
```
Comment (StrikerRUS) [inline python-package/lightgbm/sklearn.py]: Move according to the new order. Also, please place it at the same line as `callbacks`.

File: python-package/lightgbm/sklearn.py [hunk 5]
```diff
@@ -797,6 +804,7 @@ def fit(self, X, y,
                                         early_stopping_rounds=early_stopping_rounds,
                                         verbose=verbose, feature_name=feature_name,
                                         categorical_feature=categorical_feature,
+                                        init_model=init_model,
```
Comment (StrikerRUS) [inline python-package/lightgbm/sklearn.py]: Move according to the new order. Also, please place it at the same line as `callbacks`.

File: python-package/lightgbm/sklearn.py [hunk 6]
```diff
@@ -912,6 +921,7 @@ def fit(self, X, y,
                                     early_stopping_rounds=early_stopping_rounds,
                                     verbose=verbose, feature_name=feature_name,
                                     categorical_feature=categorical_feature,
+                                    init_model=init_model,
```
Comment (StrikerRUS) [inline python-package/lightgbm/sklearn.py]: Move according to the new order. Also, please place it at the same line as `callbacks`.

**Comment by StrikerRUS** [comment]
@aaiyer Please rebase to the latest `master` in order to pass CI checks.

**Comment by aaiyer** [comment]
> @aaiyer Something went wrong during the branch rebasing:
> image
> 
> Please fix it.

My bad. Fixed

**Comment by aaiyer** [comment]
> @aaiyer Please, make it possible to pass sklearn model as `init_model`.
> I guess, it can be something like
> https://github.com/microsoft/LightGBM/blob/dee72159d36a66d72372ec7fa783d1682eef1411/python-package/lightgbm/plotting.py#L88-L89

Done

File: python-package/lightgbm/sklearn.py [hunk 7]
```diff
@@ -586,13 +589,16 @@ def _get_meta_data(collection, name, i):
                                                    valid_weight, valid_init_score, valid_group, params)
                 valid_sets.append(valid_set)
 
+        if init_model is not None and isinstance(init_model, LGBMModel):
```
Comment (StrikerRUS) [inline python-package/lightgbm/sklearn.py]: Seems that the `init_model is not None` check is redundant. `isinstance(init_model, LGBMModel)` is enough.

**Comment by StrikerRUS** [comment]
Also, please update the docstring accordingly.

**Comment by StrikerRUS** [comment]
Gently ping @aaiyer

**Comment by aaiyer** [comment]
> Gently ping @aaiyer

My bad, will get to this very shortly.

**Pull request merged**